src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp

```diff
@@ -2132,6 +2133,8 @@ static int programXSpiDrv(std::FILE *mFlashDev, std::istream& mcsStream,
         if (ret)
             return ret;
         assert(nextAddr == UINT_MAX || pageOffset(nextAddr) == 0);
+        if (!storeAddr)
+            storeAddr = curAddr;
```

Comment:
If MCS initial address is "0". Your expectation is to store "0" in storeAddr right?
This logic is not going to store "0". It always stores first non-zero address. You may need to use some boolean that initial address is already stored.